src/k8s_utils.py

```diff
@@ -0,0 +1,23 @@
+"""Utility methods related to Kubernetes resources."""
+
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.resources.core_v1 import Namespace, ServiceAccount
+
+
+def is_valid_namespace(namespace: str) -> bool:
+    """Return whether given namespace exists in K8s cluster."""
+    try:
+        Client().get(Namespace, name=namespace)
+    except ApiError:
+        return False
+    return True
+
+
+def is_valid_service_account(namespace: str, service_account: str) -> bool:
```

Comment:
Here I will also check if there are enough permission to run charmed spark jobs, you can re-use some of the work we did in the spark8t library.

Reply:
I tried checking the permissions with `kubectl auth can-i --as sytem:serviceaccount:foo:bar`, however this fails inside the context of the charm unit.

The charm unit pod assumes the role of `system:serviceaccount:model_name:app_name` which is created by default by Juju, and this service account is not able to impersonate other service accounts to check RBAC with `--as` option. And as this check will always run inside the context of unit pod, I was not able to find a reliable way to check the actual permissions.

In Spark8t, we used these checks during integration tests which run in the context of edge machine (which has role of `admin` user account).

Let me know if you think there's workaround around this, I'm open for discussion.